development-guide.md

```diff
@@ -101,9 +101,14 @@ This operation should fail if it is asked to create a table that already exists.
 #### DescribeTable
 This operation should report all columns in the destination table, including Fivetran system columns such as `_fivetran_synced` and `_fivetran_deleted`. It should also provide other additional information as applicable such as data type, `primary_key` and `DecimalParams`.
 
-#### Truncate
+#### TruncateRequest
 - This operation might be requested for a table that does not exist in the destination. In that case, it should NOT fail, simply ignore the request and return `success = true`.
-- `utc_delete_before` has millisecond precision.
+- `SoftTruncate`: This optional field can be configured to either regular `soft-truncate` mode by setting the `deleted_column` field or history `soft-truncate` mode by setting the `history_mode` field to TRUE. Both modes keep the rows in the destination and update relevant system columns to mark any matching rows as deleted. If optional `SoftTruncate` field is not configured, then perform a `hard-truncate` operation where matching rows are actually removed from the destination.
+- `utc_delete_before`: This timestamp field has millisecond precision and used to DELETE/UPDATE the rows in destination table as per the following:
+  - **Soft-Truncate:** Set `deleted_column` to TRUE where `_fivetran_synced` column value is less than `utc_delete_before`.
```

Comment:
What is the actor of the set and delete actions? Is it on the user to implement or it's done automatically? In the latter case, I believe the passive voice should be a better option, like "column is set" etc.